keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+class RGBShift(layers.Layer):
+    """Randomly shift values for each channel of the input image(s).
+
+    The input images should have values in the `[0-255]` range.
+
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `channels_last` format.
+
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `channels_last` format.
+
+    Args:
+        factor: A scalar value, or tuple/list of two values. 
+            number. If `factor` is a single value, it is interpreted as 
+            equivalent to the tuple `(-factor, factor)`.
+            If the lower and upper founds are floats in the range `[-1, 1]`,
+            then the bounds are interpreted as fractions. If the lower
+            and upper bounds are integers in the range `[0, 255]`, then
+            the bounds are intepreted as absolute pixel values.
+        seed: Integer. Used to create a random seed. Default: None.
+
+    Call arguments: 
+        images: Tensor representing images of shape
+            `(..., height, width, channels)`, with dtype tf.float32 / tf.uint8, or,
+            `(height, width, channels)`, with dtype tf.float32 / tf.uint8
+        training: A boolean argument that determines whether the call should be run 
+            in inference mode or training mode. Default: True.
+   
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    rgb_shift = keras_cv.layers.RGBShift(factor=(-2, 2))
+    augmented_images = rgb_shift(images)
+    ```
+    """
+
+    _FACTOR_VALIDATION_ERROR = (
+        "The factor should be a scalar, "
+        "a tuple or a list of two upper and lower "
+        "bound values in the range `(-1.0, 1.0)` as float or `(-255, 255) as integer."
```

Comment:
lets make factor scaled to the range: `[0, 1]` and perform a transformation to map to `[-255, 255]` internally.  Then, we can use `utils.preprocessing.transform_value_range` to always operate in the [0, 255] range.

Reply:
Done.